File: Sub Palindromes.java
import java.util.*;
public class Main {
    public static void main(String[] a) {
        Scanner s = new Scanner(System.in);
        String x = s.nextLine();
        int n = x.length(), c = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                String p = x.substring(i, j + 1);
                String r = new StringBuilder(p).reverse().toString();
                if (p.equals(r)) c++;
            }
        }
        System.out.println(c);
    }
}

Example 1:
Input:
everest
Output:
2
Example 2:
Input:
abccbaab
Output:
5

File: Sub Palindromes.cpp
#include <iostream>
#include <string>
using namespace std;

bool isPalindrome(const string& s) {
    int i = 0, j = s.length() - 1;
    while (i < j) {
        if (s[i] != s[j]) {
            return false;
        }
        i++;
        j--;
    }
    return true;
}

int main() {
    string s;
    cin >> s;

    int n = s.length(), count = 0;

    // Check all substrings of length >= 2
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            string sub = s.substr(i, j - i + 1);  // Extract substring
            if (sub.length() >= 2 && isPalindrome(sub)) {
                count++;
            }
        }
    }

    cout << count << endl;  // Output the count of sub palindromes
    return 0;
}

Example 1:
Input:
everest
Output:
2
Example 2:
Input:
abccbaab
Output:
5
